code/ndarray.c

```diff
@@ -164,16 +164,19 @@ STATIC uint8_t ndarray_init_helper(size_t n_args, const mp_obj_t *pos_args, mp_m
     return dtype;
 }
 
-mp_obj_t ndarray_make_new(const mp_obj_type_t *type, size_t n_args, size_t n_kw, const mp_obj_t *args) {
-    mp_arg_check_num(n_args, n_kw, 1, 2, true);
-    mp_map_t kw_args;
-    mp_map_init_fixed_table(&kw_args, n_kw, args + n_args);
-    uint8_t dtype = ndarray_init_helper(n_args, args, &kw_args);
+mp_obj_t ndarray_make_new(const mp_obj_type_t *type, size_t n_args, const mp_obj_t *args, mp_map_t *kw_args) {
```

Comment:
I think this won't compile under micropython, so we really need a switching mechanism for this case.

Reply:
Yes, we (CircuitPython) changed the way `...make_new()` is called to accommodate something (I think it may have been better support of Python subclasses of native classes). Damien was supportive of this but felt he could not change the current MicroPython internal API due to so many modules depending on it.

Note that we have a preprocessor value that can be tested to determine if it's CircuitPython.
```c
#if CIRCUITPY
...
```
Also note we in general try to use `#if` tests rather than `#ifdef` because they're safer, and we also usually turn on the warning that will complain if `FOO` is undefined for
`#if FOO`

Reply:
OK, I can go with that, no problem.